### ISSUE TITLE
TypeError: Request with a GET or HEAD method cannot have a body.

### ISSUE THREAD

**Issue by eftikharEmad** [Bandwidth/ruby-sdk#43]
Hello Team, 

we are using bandwidth for over 3 years now, the configuration is 
```
BandwidthClient = Bandwidth::Client.new(
  voice_basic_auth_user_name: settings.username,
  voice_basic_auth_password: settings.password,
  messaging_basic_auth_user_name: settings.messages.api_token,
  messaging_basic_auth_password: settings.messages.api_secret,
)
```
for any call related operation like play a call recording should be via the voice_client so we are using such client variable 
```
@voice_client = BandwidthClient.voice_client.client
```

and the method to list the call recordings, is simply 
```
def recordings(call_id)
    response = @voice_client.get_query_metadata_for_account_and_call(@settings.account_id, call_id)
end
```

this method was working fine, however recently I was getting this error 

```
TypeError: Server responded with invalid JSON
```

I went through the gem version code I have and add some tracking statements for this method `get_query_metadata_for_account_and_call` see what I got 

**this is the method**
```
 def get_query_metadata_for_account_and_call(account_id,
                                                call_id,
                                                from: nil,
                                                to: nil,
                                                min_start_time: nil,
                                                max_start_time: nil)
      # Prepare query url.
      _query_builder = config.get_base_uri(Server::VOICEDEFAULT)
      _query_builder << '/api/v2/accounts/{accountId}/calls/{callId}/recordings'
      _query_builder = APIHelper.append_url_with_template_parameters(
        _query_builder,
        'accountId' => account_id,
        'callId' => call_id
      )
      _query_builder = APIHelper.append_url_with_query_parameters(
        _query_builder,
        'from' => from,
        'to' => to,
        'minStartTime' => min_start_time,
        'maxStartTime' => max_start_time
      )
      _query_url = APIHelper.clean_url _query_builder

      # Prepare headers.
      _headers = {
        'accept' => 'application/json'
      }

      # Prepare and execute HttpRequest.
      puts "config: #{config}"
      puts "_query_url: #{_query_url}"
      puts "_headers: #{_headers}"
      _request = config.http_client.get(
        _query_url,
        headers: _headers
      )
      puts "_request: #{_request}"
      VoiceBasicAuth.apply(config, _request)
      _response = execute_request(_request)
      puts "_response.status_code: #{_response.status_code}, _response: #{_response.raw_body}"

      # Validate response against endpoint and global error codes.
      if _response.status_code == 400
        raise ApiErrorResponseException.new(
          'Something\'s not quite right... Either your request is invalid or' \
          ' you\'re requesting it at a bad time. Please fix it' \
          ' before trying again.',
          _response
        )
.
.
.
```
**Bandwidth::Voice::APIController**

```
 => #<Bandwidth::Voice::APIController:0x007fc00e842d28 @config=#<Bandwidth::Configuration:0x007fc029535700 @timeout=60, @max_retries=0, @retry_interval=1, @backoff_factor=1, @environment="production", @messaging_basic_auth_user_name="........", @messaging_basic_auth_password="........", @two_factor_auth_basic_auth_user_name="TODO: Replace", @two_factor_auth_basic_auth_password="TODO: Replace", @voice_basic_auth_user_name=".....", @voice_basic_auth_password="......", @http_client=#<Bandwidth::FaradayClient:0x007fc0295350c0 @connection=#<Faraday::Connection:0x007fc029534e18 @parallel_manager=nil, @headers={"User-Agent"=>"Faraday v0.17.3"}, @params={}, @options=#<Faraday::RequestOptions params_encoder=Faraday::FlatParamsEncoder, timeout=60>, @ssl=#<Faraday::SSLOptions verify=true, ca_file="/Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/certifi-2018.01.18/lib/certifi/vendor/cacert.pem">, @default_parallel_manager=nil, @builder=#<Faraday::RackBuilder:0x007fc029534b20 @handlers=[FaradayMiddleware::FollowRedirects, Faraday::Request::Multipart, Faraday::Request::UrlEncoded, Faraday::Request::Retry, Faraday::Adapter::NetHttp]>, @url_prefix=#<URI::HTTP http:/>, @manual_proxy=false, @proxy=nil, @temp_proxy=nil>>>, @http_call_back=nil, @global_headers={"user-agent"=>"ruby-sdk-refs/tags/ruby3.3.0"}>
```

**AND THIS IS THE EXCEPTION TRACE** 
```
config: #<Bandwidth::Configuration:0x007fc029535700>
_query_url: https://voice.bandwidth.com/api/v2/accounts/******/calls/c-28bba681-3804721e-1605-4462-b4bb-712bb02e72f8/recordings
_headers: {"accept"=>"application/json"}
_request: #<Bandwidth::HttpRequest:0x007fc025bc42a0>
_response.status_code: 500, _response: TypeError: Request with a GET or HEAD method cannot have a body.
    at handleRequest (worker.js:41:33)
    at worker.js:55:5
TypeError: Server responded with invalid JSON.......:TypeError: Request with a GET or HEAD method cannot have a body.
    at handleRequest (worker.js:41:33)
    at worker.js:55:5
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/bandwidth-sdk-3.3.0/lib/bandwidth/api_helper.rb:142:in `rescue in json_deserialize'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/bandwidth-sdk-3.3.0/lib/bandwidth/api_helper.rb:140:in `json_deserialize'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/bandwidth-sdk-3.3.0/lib/bandwidth/voice_lib/voice/exceptions/api_error_response_exception.rb:26:in `initialize'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/bandwidth-sdk-3.3.0/lib/bandwidth/voice_lib/voice/controllers/api_controller.rb:418:in `new'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/bandwidth-sdk-3.3.0/lib/bandwidth/voice_lib/voice/controllers/api_controller.rb:418:in `get_query_metadata_for_account_and_call'
	from (irb):7
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7/lib/rails/commands/console.rb:112:in `start'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7/lib/rails/commands/console.rb:11:in `start'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/eftikharabulibdeh/projects/backend/vendor/bundle/ruby/2.3.0/gems/railties-4.2.7/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
```

as you can see I am using `bandwidth-sdk-3.3.0`, can you please advice me if I can do any thing from my side because I think there is something I am missing

**Comment by ajrice6713** [comment]
Hey @eftikharEmad 

Ive seen this come up with some other customers and I believe it is an issue with our backend and not the request being sent by the SDK. Have you opened a ticket on support.bandwidth.com for this issue?

**Comment by ckoegel** [comment]
Hi @eftikharEmad
This was an issue we identified a little while ago and it has been fixed in the latest version of the SDK (9.1.1). Updating to the latest version should fix your issue, or if you can't update, changing [this](https://github.com/Bandwidth/ruby-sdk/blob/8e93b9ae085436e46c06e825d0afc536934afc86/lib/bandwidth/http/faraday_client.rb#L37) line to `request.params = http_request.parameters` in your version should solve it.

**Comment by eftikharEmad** [comment]
Hello @ajrice6713 @ckoegel 
Thank you for replying, issue solved by upgrading gem to latest version, we found a work around solution, and we are good.
thank you so much
